### PR TITLE
Add page for plugin configuration

### PR DESCRIPTION
**Pull request by d2lam**
Adding some documentation for configuring plugins. 
Relates to https://github.com/screwdriver-cd/screwdriver/issues/304

screen shot 2016-11-01 at 1 55 57 pm

### CODE REVIEW

File: docs/getting-started/plugins.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+datastore:
+    plugin: dynamodb
+    dynamodb:
+        # Prefix to the table names
+        prefix: TABLE-PREFIX
+        accessKeyId: AWS-ACCESS-KEY-ID
+        secretAccessKey: AWS-SECRET-ACCESS-KEY
+```
+
+Or set these environment variables in your `docker-compose.yml`:
```
Comment (minzcmu) [inline docs/getting-started/plugins.md]: There is no `docker-compose.yml` if you don't run `in-a-box`. Can we make it clear here?

File: docs/getting-started/plugins.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+datastore:
+    plugin: sequelize
+    sequelize:
+        dialect: TYPE-OF-SERVER           # Ex: "sqlite"
+        storage: STORAGE-LOCATION         # Ex: "./database.sqlite"
+        database: DATABASE-NAME
+        username: DATABASE-USERNAME
+        password: DATABASE-PASSWORD
+        host: NETWORK-HOST
+        port: NETWORK-PORT
```
Comment (minzcmu) [inline docs/getting-started/plugins.md:53]: Can we make it clear here what is required and what is optional?

File: docs/getting-started/plugins.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+## Source Control
+
+### Step 1: Set up your OAuth Application
+You will need to set up an OAuth Application and retrieve your OAuth Client ID and Secret.
+
+#### Github:
+1. Navigate to the [Github OAuth applications](https://github.com/settings/developers) page.
+2. Click on the application you created to get your OAuth Client ID and Secret.
+3. Fill out the `Homepage URL` and `Authorization callback URL` with your `LoadBalancer Ingress`.
+
```
Comment (minzcmu) [inline docs/getting-started/plugins.md:110]: How about running it locally? You don't really need `LoadBalancer Ingress`, right?

**Review by minzcmu — CHANGES_REQUESTED** [review]

File: docs/getting-started/plugins.md (hunk 1 above)
Comment (FenrirUnbound) [reply docs/getting-started/plugins.md]: Alternatively, you can clarify that the `docker-compose.yml` is created by `in-a-box`

File: docs/getting-started/plugins.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
+# Configure your plugins
+Screwdriver already defaults most configuration, but you can override defaults using a `local.yaml` or environment variables.
+
+## Datastore
+To use DynamoDB, use `dynamodb` plugin. To use Postgres, MySQL, and Sqlite, use `sequelize` plugin.
```
Comment (FenrirUnbound) [inline docs/getting-started/plugins.md]: Might be useful to provide a link to the respective modules

File: docs/getting-started/plugins.md (hunk 3 above)
Comment (petey) [reply docs/getting-started/plugins.md:110]: Nothing on this page explains what a `LoadBalancer Ingress` is.

File: docs/getting-started/plugins.md (hunk 2 above)
Comment (d2lam) [reply docs/getting-started/plugins.md:53]: Fixed. Switching to tables so we can squeeze in more fields.

File: docs/getting-started/plugins.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+WEBHOOK_GITHUB_SECRET: SUPER-SECRET-SIGNING-THING
+SCM_GITHUB_GHE_HOST: GHE-HOST
+```
+
+#### Bitbucket.org
+Modify your `local.yaml`:
+```yaml
+scm:
+    plugin: bitbucket
+    github:
```
Comment (tkyi) [inline docs/getting-started/plugins.md]: `bitbucket:`

File: docs/getting-started/plugins.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+        # gheHost: github.screwdriver.cd
+```
+
+Or set these environment variables in your `docker-compose.yml`:
+```yaml
+SCM_PLUGIN: github
+SECRET_OAUTH_CLIENT_ID: YOUR-OAUTH-CLIENT-SECRET
+SECRET_OAUTH_CLIENT_SECRET: YOUR-OAUTH-CLIENT-ID
+WEBHOOK_GITHUB_SECRET: SUPER-SECRET-SIGNING-THING
+SCM_GITHUB_GHE_HOST: GHE-HOST
```
Comment (tkyi) [inline docs/getting-started/plugins.md]: Can we say add a comment saying this is optional?

**Review by tkyi — CHANGES_REQUESTED** [review]

File: docs/getting-started/plugins.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+        secret: SUPER-SECRET-SIGNING-THING
+        # You can also configure for use with GitHub enterprise
+        # gheHost: github.screwdriver.cd
+```
+
+#### Bitbucket.org
+```yaml
+scm:
+    plugin: bitbucket
+    github:
```
Comment (tkyi) [inline docs/getting-started/plugins.md]: bitbucket:

File: docs/getting-started/plugins.md [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+| Environment name | Required | Default Value | Description |
+| :--- | :---- | :--- | :--- | :--- |
+| DATASTORE_PLUGIN | Yes| | Set to `sequelize`
+| DATASTORE_SEQUELIZE_DIALECT | No | mysql | Can be `sqlite`, `postgres`, `mysql`, or `mssql`
+| DATASTORE_SEQUELIZE_DATABASE | No | screwdriver | Database name
+| DATASTORE_SEQUELIZE_USERNAME | No for sqlite | | Login username
+| DATASTORE_SEQUELIZE_PASSWORD | No for sqlite | | Login password
+| DATASTORE_SEQUELIZE_STORAGE | Yes for sqlite | | Storage location for sqlite
+| DATASTORE_SEQUELIZE_HOST | No | | Network host
+| DATASTORE_SEQUELIZE_PORT | No | | Network host
```
Comment (minzcmu) [inline docs/getting-started/plugins.md]: network port?

**Review by tkyi — APPROVED** [review]

**Review by minzcmu — APPROVED** [review]
The rest LGTM 👍

**Pull request merged**